### PR TITLE
cardano-cli:  improve the help facilities and the README

### PR DESCRIPTION
**Pull request by deepfire**
1. Arg parsing: make the mode less opaque by switching to an option, hide currently unused system versions.  Update scripts to the changes.
2. Improve the README.

### CODE REVIEW

File: cardano-node/README.md [hunk 1]
```diff
@@ -116,25 +114,36 @@ Available options:
...
 
-The canned `scripts/genesis.sh` example provides a nice set of defaults and
-illustrates available options.  Running it will produce a `./genesis.${systemStart}` directory,
-where `${systemStart}` will be chosen 15 minutes in the future.
+### Generation
+
+The genesis generation operations will create a directory that contains:
 
-The directory will have the following content:
+  - the genesis JSON file itself (`genesis.json`)
```
Comment (Jimbo4350) [inline cardano-node/README.md]: Can we document each of the items it creates? e.g Which keys are generated and what do they do etc. Doesn't have to be long. You could link to other documentation that explains this if possible. I'm not sure if this is up to date: https://cardanodocs.com/

File: cardano-node/src/Cardano/Common/Parsers.hs [hunk 2]
```diff
@@ -46,19 +47,31 @@ parseNodeId desc =
...
   , fl MockPBFT "mock-pbft"
-    "Use the Permissive BFT consensus algorithm using a mock ledger"
+    "Permissive BFT consensus with a mock ledger"
 
   , fl RealPBFT "real-pbft"
-    "Use the Permissive BFT consensus algorithm using the real ledger"
+    "Permissive BFT consensus with a real ledger"
+  ]
+  where
+    fl :: forall a. a -> String -> String -> Parser a
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Common/Parsers.hs]: `fl` is duplicated below. Factoring out with a comment would be nice :).

Comment (intricate) [reply cardano-node/src/Cardano/Common/Parsers.hs]: @deepfire: What does `fl` stand for?

Comment (deepfire) [reply cardano-node/src/Cardano/Common/Parsers.hs]: Done.

File: cardano-node/README.md (hunk 1 above)
Comment (deepfire) [reply cardano-node/README.md]: Done.

**Pull request merged**